Sources/FoundationEssentials/Calendar/Calendar.swift

```diff
@@ -733,8 +733,8 @@ public struct Calendar : Hashable, Equatable, Sendable {
                 return .orderedAscending
             }
         case .second:
-            let (int1, _) = modf(date1.timeIntervalSinceReferenceDate)
-            let (int2, _) = modf(date2.timeIntervalSinceReferenceDate)
+            let int1 = date1.timeIntervalSinceReferenceDate.rounded(.down)
+            let int2 = date2.timeIntervalSinceReferenceDate.rounded(.down)
```

Comment:
Oh I just realized: do we want .towardZero here too?

Reply:
(This would be a behavior change from the old code, I'm just wondering if it's more correct in edge cases)

Reply:
Hm, I'll need to think about it separately.